src/target/llvm/codegen_arm.cc

```diff
@@ -55,7 +55,13 @@ class CodeGenARM final : public CodeGenCPU {
 
 llvm::Value* CodeGenARM::CreateIntrinsic(const CallNode* op) {
   if (op->op.same_as(builtin_call_llvm_intrin_) || op->op.same_as(builtin_call_llvm_pure_intrin_)) {
-    llvm::Intrinsic::ID id = static_cast<llvm::Intrinsic::ID>(Downcast<IntImm>(op->args[0])->value);
+    llvm::Intrinsic::ID id = 0;
+    if (op->args[0]->IsInstance<StringImmNode>()) {
+      id = llvm::Function::lookupIntrinsicID(Downcast<StringImm>(op->args[0])->value.c_str());
+    } else if (op->args[0]->IsInstance<IntImmNode>()) {
+      id = static_cast<llvm::Intrinsic::ID>(Downcast<IntImm>(op->args[0])->value);
+    }
+    assert(id != 0);
```

Comment:
Removed the check instead.

1. Actually ```id=0``` exists in LLVM internals.
  ```
  DBG [not_intrinsic] (0)
  DBG [llvm.abs] (1)
  DBG [llvm.addressofreturnaddress] (2)
  DBG [llvm.adjust.trampoline] (3)
  DBG [llvm.annotation] (4)
  ```
2. From high level python `op.py` the ```codegen.llvm_get_intrinsic_name(llvm_id)``` would fail on bad user input.

Reply:
Thanks for the clarification. I agree that this seems logical, however I think it might have been better to keep the check as now the code has an `if - else if` without an else or assertion.

Without the assertion, we're assuming that a call to the python API is the only way the LLVM intrinsic would be inserted, but what if a pass inserts the intrinsic call from cpp directly?

Reply:
Let's protect both: node type in falling ```else``` and the ```id !=0``` (via C api).
Added a testcase to induce a invalid id=0.

Reply:
Sounds good. Thanks for taking my feedback.